🔧 Fix(Fragments/Leptos): Fix unwanted refresh and update to Leptos v0.2

pikhosh opened the pull request:
<!--
Update "[ ]" to "[x]" to check a box

Please make sure to read the Pull Request Guidelines: https://github.com/tauri-apps/tauri/blob/dev/.github/CONTRIBUTING.md#pull-request-guidelines
-->

### What kind of change does this PR introduce?
<!-- Check at least one. If you are introducing a new binding, you must reference an issue where this binding has been proposed, discussed and approved by the maintainers. -->

- [x] Bugfix
- [ ] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?
<!-- If yes, please describe the impact and migration path for existing applications in an attached issue. -->

- [ ] Yes, and the changes were approved in issue #___
- [x] No

### Checklist
- [ ] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [x] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/create-tauri-app/blob/dev/.changes/readme.md).
- [ ] I have added a convincing reason for adding this feature, if necessary

### Other information
Nothing...

pikhosh committed:
🔧 Fix(Fragments/Leptos): Fix unwanted refresh when clicking on the button

Basically just a `preventDefault()` :)

pikhosh committed:
🗿 Chore(Fragments/Leptos): Update to Leptos v0.2.1

pikhosh committed:
🗿 Chore: Create changes.md

I don't know if this is needed or not... but anyway!

amrbashir commented on packages/cli/fragments/fragment-leptos/_Cargo.toml:
```diff
@@ -5,11 +5,13 @@ edition = "2021"
-serde-wasm-bindgen = "0.4"
-wasm-bindgen = { version = "0.2", features = ["serde-serialize"] }
-wasm-bindgen-futures = "0.4"
+leptos = { version = "0.2.1", features = ["stable"] }
+serde = { version = "1.0.153", features = ["derive"] }
+serde-wasm-bindgen = "0.5.0"
+wasm-bindgen = { version = "0.2.84", features = ["serde-serialize"] }
+wasm-bindgen-futures = "0.4.34"
+js-sys = "0.3.61"
+web-sys = "0.3.61"
```
```suggestion
leptos = { version = "0.2", features = ["stable"] }
serde = { version = "1.0", features = ["derive"] }
serde-wasm-bindgen = "0.5"
wasm-bindgen = { version = "0.2", features = ["serde-serialize"] }
wasm-bindgen-futures = "0.4"
js-sys = "0.3"
web-sys = "0.3"
```

amrbashir commented on packages/cli/fragments/fragment-leptos/src/app.rs:
```diff
@@ -60,7 +60,10 @@ pub fn App(cx: Scope) -> impl IntoView {
             </p>
 
-            <form class="row" on:submit=greet>
+            <form class="row" on:submit=move |ev: SubmitEvent| {
+                ev.prevent_default()
+                greet()
+            }>
```
is there a reason behind this change? I think it should be reverted to have consistency with other templates and just call `ev.prevent_default()` above inside the closure

pikhosh replied:
Ummm... I don't know why I didn't think of that... yeah that makes more sense... :|

amrbashir commented on .changes/changes.md:
```diff
@@ -0,0 +1,6 @@
+"create-tauri-app": patch
+---
+
+Fix(Fragments/Leptos): Fix unwanted refresh when clicking on the button
+Chore(Fragments/Leptos): Update to Leptos v0.2.1
```
```suggestion
Fix unwanted refresh when clicking on the greet button in `leptos` template
```
and add `Update Leptos to v0.2` in another change file

amrbashir reviewed (changes requested):
thank you

pikhosh committed:
Update packages/cli/fragments/fragment-leptos/_Cargo.toml

Co-authored-by: Amr Bashir <amr.bashir2015@gmail.com>

pikhosh committed:
Update app.rs

pikhosh committed:
Merge branch 'leptos-update' of https://github.com/pikhosh/create-tauri-app into leptos-update

pikhosh committed:
Update .changes/changes.md

Co-authored-by: Amr Bashir <amr.bashir2015@gmail.com>

amrbashir commented on packages/cli/fragments/fragment-leptos/_Cargo.toml:
```diff
@@ -5,11 +5,13 @@ edition = "2021"
+serde-wasm-bindgen = "0.5"
 wasm-bindgen = { version = "0.2", features = ["serde-serialize"] }
 wasm-bindgen-futures = "0.4"
+js-sys = "0.3"
+web-sys = "0.3"
```
I don't think these are needed

pikhosh replied:
`wasm_bindgen` need `js-sys`.

amrbashir replied:
keep `js-sys` then and remove `web-sys` if it is not needed

amrbashir commented on packages/cli/fragments/fragment-leptos/src/app.rs:
```diff
@@ -1,4 +1,4 @@
-use leptos::web_sys::{Event, SubmitEvent};
+use web_sys::{Event, SubmitEvent};
```
```suggestion
use leptos::leptos_dom::ev::{InputEvent, SubmitEvent};
```
and update the `update_name` closure to take `InputEvent` instead

pikhosh replied:
Apparently `on:input` event type is `Event` and I get an error when using `InputEvent`...

amrbashir replied:
Seems like leptos v0.2 is missing a re-export of `Event` type, Remove the `Event` import all together and change the on:input handler to 
```rs
on:input=move |ev| {
  let v = event_target_value(&ev);
  update_name(v)
}
```
then update `update_name` function to take `String` type instead

pikhosh replied:
Why not just remove the type from the `update_name` closure?

amrbashir replied:
if it works, yes but last time I created the template it wasn't able to infer the type.

pikhosh replied:
Works for me.

amrbashir replied:
cool, lets go with that then

pikhosh committed:
🗿 Chore: Add all necessary files to the `.changes` folder

Co-Authored-By: Amr Bashir <amr.bashir2015@gmail.com>

pikhosh committed:
🛠️ Refactor(Fragments/Leptos): Remove `web-sys`

Leptos v0.2 doesn't re-export the `Event` type from the `web-sys` so in order to remove the `web-sys` we need to remove the explicit type from the `update_name` closure

Co-Authored-By: Amr Bashir <amr.bashir2015@gmail.com>

amrbashir reviewed (approved):
Thank you
